WordPress/Classes/Services/Store.swift

```diff
@@ -22,53 +22,162 @@ class StoreKitTransactionObserver: NSObject, SKPaymentTransactionObserver {
+    static let NotificationProductIdentifierKey = "StoreCoordinatorNotificationProductIdentifierKey"
 }
 
+typealias PendingPayment = (planID: PlanID, productID: String, siteID: Int)
+
+enum StoreCoordinatorError: ErrorType {
+    case PaymentAlreadyInProgress
+}
+
+/// StoreCoordinator coordinates purchasing of products, processing of transactions, and
+/// verification of purchases between the App Store (StoreKit) and WordPress.com.
+///
+/// Users of this class can simply attempt a purchase using `purchasePlan(_:product:forSite)`,
+/// and the coordinator will post a notification on a successful or failed purchase:
+///
+/// - `StoreKitCoordinator.TransactionDidFinishNotification` on success.
+///   The notification's `userInfo` will contain the productID of the purchased
+///   product under `StoreKitCoordinator.NotificationProductIdentifierKey`.
+/// - `StoreKitCoordinator.TransactionDidFailNotification` on failure.
+///   The notification's `userInfo` will also contain the productID of the attempted
+///   purchased product, as well as a localized error message under `NSUnderlyingErrorKey`.
 class StoreCoordinator<S: Store> {
-    let store: S
-    var pendingPayment: (plan: Plan, siteID: Int)? = nil
+    private let store: S
+    
+    private var pendingPayment: PendingPayment? {
+        set {
+            let defaults = NSUserDefaults.standardUserDefaults()
+            
+            if let pending = newValue {
+                defaults.setInteger(pending.planID, forKey: UserDefaultsKeys.pendingPaymentPlanID)
+                defaults.setObject(pending.productID, forKey: UserDefaultsKeys.pendingPaymentProductID)
+                defaults.setInteger(pending.siteID, forKey:UserDefaultsKeys.pendingPaymentSiteID)
+            } else {
+                defaults.removeObjectForKey(UserDefaultsKeys.pendingPaymentPlanID)
+                defaults.removeObjectForKey(UserDefaultsKeys.pendingPaymentProductID)
+                defaults.removeObjectForKey(UserDefaultsKeys.pendingPaymentSiteID)
+            }
+        }
+        
+        get {
+            let defaults = NSUserDefaults.standardUserDefaults()
+            let planID = defaults.integerForKey(UserDefaultsKeys.pendingPaymentPlanID)
+            let productID = defaults.stringForKey(UserDefaultsKeys.pendingPaymentProductID)
+            let siteID = defaults.integerForKey(UserDefaultsKeys.pendingPaymentSiteID)
+            
+            guard let product = productID where planID != 0 && siteID != 0 else { return nil }
+            
+            return (planID, product, siteID)
+        }
+    }
 
     init(store: S) {
         self.store = store
     }
 
-    func purchasePlan(plan: Plan, product: S.ProductType, forSite siteID: Int) {
+    /// Initiates a purchase for the specified plan if a purchase isn't already in progress.
+    ///
+    /// - Throws: A `StoreCoordinatorError.PaymentAlreadyInProgress` error if the purchase 
```

Comment:
Minor style, the [Markup Formatting Reference](https://developer.apple.com/library/ios/documentation/Xcode/Reference/xcode_markup_formatting_ref/Throws.html#//apple_ref/doc/uid/TP40016497-CH26-SW1) uses lowercase for all their callout examples.